app/controllers/cards_controller.rb

```diff
@@ -2,4 +2,44 @@ class CardsController < ApplicationController
   def index
     @cards = Card.all.order(original_text: :asc)
   end
+
+  def new
+    @card = Card.new
+  end
+
+  def create
+    @card = Card.new(card_params)
+    
+    if @card.save
+      redirect_to cards_path
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    @card = Card.find(params[:id])
+  end
+
+  def update
+    @card = Card.find(params[:id])
+    
+    if @card.update(card_params)
+      redirect_to cards_path
+    else
+      render 'edit'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.